wrappers/binaries.go

```diff
@@ -45,7 +103,16 @@ func AddSnapBinaries(s *snap.Info) (err error) {
 	}
 
 	completeSh := dirs.CompleteShPath(s.Base)
-	noCompletion := !osutil.IsWritable(dirs.CompletersDir) || !osutil.FileExists(completeSh)
+	noCompletion := !osutil.FileExists(completeSh)
```

Comment:
it sounds like this could be moved inside detectLegacy() (if we rename it to detectCompletionSupport and make it return an enum)

Reply:
Done